### DISCUSSION TITLE
use of undeclared crate or module `eframe`

### DISCUSSION THREAD

**Discussion by chlin501** [emilk/egui#4533] [General]
I am very newbie to egui. I want to use egui for my side project. Searching the discussion here, but I do not find any related [discussion](https://github.com/emilk/egui/discussions?discussions_q=use+of+undeclared+crate+or+module+%60eframe%60). After adding egui crate to my project in Cargo.toml

```
[package]
name = "myapp"
version = "0.1.0"
edition = "2021"

[dependencies]
...
egui = "0.27.2"
```

In the **main.rs** file, I add `use eframe::egui;`. Then building the app with `cargo build`. Then the rust compiler complains 
```
error[E0432]: unresolved import `eframe`
 --> src/main.rs:8:5
  |
8 | use eframe::egui;
  |     ^^^^^^ use of undeclared crate or module `eframe`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `myapp` (bin "myapp") due to 1 previous error
```

How should I fix this error? Thanks

**Comment by YgorSouza** [comment]
If you are using eframe, you should add eframe as a dependency, not egui. See these sections from the README:

<https://github.com/emilk/egui?tab=readme-ov-file#what-is-the-difference-between-egui-and-eframe>
<https://github.com/emilk/egui?tab=readme-ov-file#quick-start>

**Comment by chlin501** [comment]
Oh did not know that. This fixes my problem. Thank you!
